### PR TITLE
fix: making tenantId optional

### PR DESCRIPTION
**Pull request by sattvikc**
## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: recipe/thirdparty/api/providersForTenant.go [hunk 1]
```diff
@@ -26,10 +26,11 @@ func ProvidersForTenantAPI(apiImplementation tpmodels.APIInterface, options tpmo
...
 	queryParams := options.Req.URL.Query()
-	tenantId := queryParams.Get("tenantId")
+	var tenantId *string = nil
+	tenantIdStr := queryParams.Get("tenantId")
 
-	if tenantId == "" {
-		tenantId = tpmodels.DefaultTenantId
+	if tenantIdStr == "" {
+		tenantId = &tenantIdStr
 	}
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/api/providersForTenant.go:34]: Done

File: recipe/thirdparty/tpmodels/recipeInterface.go [hunk 2]
```diff
@@ -21,13 +21,13 @@ type RecipeInterface struct {
...
-	FetchTenantIdConfigMapping          *func(thirdPartyId string, tenantId string, userContext supertokens.UserContext) (FetchTenantIdConfigResponse, error)
-	DeleteTenantIdConfigMapping         *func(thirdPartyId string, tenantId string, userContext supertokens.UserContext) (DeleteTenantIdConfigResponse, error)
-	ListConfigMappingsForTenant         *func(tenantId string, userContext supertokens.UserContext) (ListTenantConfigMappingsResponse, error)
+	CreateOrUpdateTenantIdConfigMapping *func(thirdPartyId string, tenantId *string, config ProviderConfig, userContext supertokens.UserContext) (CreateOrUpdateTenantIdConfigResponse, error)
+	FetchTenantIdConfigMapping          *func(thirdPartyId string, tenantId *string, userContext supertokens.UserContext) (FetchTenantIdConfigResponse, error)
+	DeleteTenantIdConfigMapping         *func(thirdPartyId string, tenantId *string, userContext supertokens.UserContext) (DeleteTenantIdConfigResponse, error)
+	ListConfigMappingsForTenant         *func(tenantId *string, userContext supertokens.UserContext) (ListTenantConfigMappingsResponse, error)
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/tpmodels/recipeInterface.go:30]: Done

File: recipe/thirdparty/tpmodels/models.go [hunk 3]
```diff
@@ -159,18 +159,18 @@ type ProviderConfigForClientType struct {
...
-	GetAllClientTypeConfigForTenant func(tenantId string, recipeImpl RecipeInterface, userContext supertokens.UserContext) (ProviderConfig, error)
+	GetAllClientTypeConfigForTenant func(tenantId *string, recipeImpl RecipeInterface, userContext supertokens.UserContext) (ProviderConfig, error)
 	GetConfigForClientType          func(clientType *string, input ProviderConfig, userContext supertokens.UserContext) (ProviderConfigForClientType, error)
 	GetAuthorisationRedirectURL     func(config ProviderConfigForClientType, redirectURIOnProviderDashboard string, userContext supertokens.UserContext) (TypeAuthorisationRedirect, error)
 	ExchangeAuthCodeForOAuthTokens  func(config ProviderConfigForClientType, redirectURIInfo TypeRedirectURIInfo, userContext supertokens.UserContext) (TypeOAuthTokens, error) // For apple, add userInfo from callbackInfo to oAuthTOkens
 	GetUserInfo                     func(config ProviderConfigForClientType, oAuthTokens TypeOAuthTokens, userContext supertokens.UserContext) (TypeUserInfo, error)
 }
 
-const DefaultTenantId string = "default"
+const DefaultTenantId string = "public" // TODO set the value based on what's decided for core
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/tpmodels/models.go:176]: Done

File: recipe/thirdparty/tpmodels/apiInterface.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ type APIInterface struct {
 	SignInUpPOST        *func(provider TypeProvider, config ProviderConfigForClientType, input TypeSignInUpInput, options APIOptions, userContext supertokens.UserContext) (SignInUpPOSTResponse, error)
 
 	AppleRedirectHandlerPOST *func(formPostInfoFromProvider map[string]interface{}, options APIOptions, userContext supertokens.UserContext) error
-	ProvidersForTenantGET    *func(tenantId string, options APIOptions, userContext supertokens.UserContext) (ProvidersForTenantGetResponse, error)
+	ProvidersForTenantGET    *func(tenantId *string, options APIOptions, userContext supertokens.UserContext) (ProvidersForTenantGetResponse, error)
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/tpmodels/apiInterface.go:31]: Done

File: recipe/thirdparty/recipe.go [hunk 5]
```diff
@@ -104,6 +104,10 @@ func (r *Recipe) getAPIsHandled() ([]supertokens.APIHandled, error) {
 	if err != nil {
 		return nil, err
 	}
+	providersForTenantAPI, err := supertokens.NewNormalisedURLPath(ProvidersForTenantAPI)
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/recipe.go:107]: Done

File: recipe/thirdparty/providers/custom.go [hunk 6]
```diff
@@ -169,6 +169,10 @@ func NewProvider(input tpmodels.ProviderInput) tpmodels.TypeProvider {
 			return getProviderConfigForClient(inputConfig, inputConfig.Clients[0]), nil
 		}
 
+		if len(inputConfig.Clients) == 1 {
+			return getProviderConfigForClient(inputConfig, inputConfig.Clients[0]), nil
+		}
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply recipe/thirdparty/providers/custom.go:174]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**